src/message/fetchActions.js

```diff
@@ -168,8 +168,8 @@ export const fetchEssentialInitialData = () => async (dispatch: Dispatch, getSta
   timing.end('Essential server data');
 
   dispatch(realmInit(initData));
-  if (narrow && messages) {
-    dispatch(messageFetchComplete(messages, narrow, 0, halfCount, halfCount));
+  if (narrow && messagesResponse) {
```

Comment:
Good catch! I didn't consider that this is (potentially) changing the behavior.

But actually, the initial behavior is not really correct. It only checks for `messages` being `undefined` (it is true if `messages` is an empty array). But `tryUntilSuccessful` would not stop retrying until successful. If so it will always return `truthy` object.

The correct change is to remove the second check altogether.